fuel-tx/src/transaction/id.rs

```diff
@@ -108,6 +108,7 @@ where
                 }) if owner == &pk => Some(*witness_index as usize),
                 _ => None,
             })
+            .sorted()
```

Comment:
personal question: what is the reason behind this ?

Reply:
Just notice going over the code that we call `dedup`, but it not we expect=) So I added `dedup` to have behavior that we want

Reply:
Yeah, `dedup()` works on "[consecutive identical elements](https://docs.rs/itertools/latest/itertools/trait.Itertools.html#method.dedup)". This bit me many times in the past :)
Good catch.